conda/plugins/manager.py

```diff
@@ -159,6 +161,24 @@ def get_solver_backend(self, name: str = None) -> type[Solver]:
 
         return backend
 
+    def run_pre_command_hooks(self, command: str, args: Namespace) -> None:
+        """
+        Runs ``CondaPreCommand.action`` functions registered by the ``conda_pre_commands`` hook.
+
+        This method takes special care to handle any exceptions raised during the running of these
+        hooks so that it doesn't interfere with the normal operation of the conda command.
```

Comment:
Still true?

Reply:
Good catch! Not true anymore.

Reply:
OTOH it is a little tricky if you've installed something that breaks the installer, preventing the uninstall of that thing. Do we have a "no plugins" safe mode flag?

Reply:
This was what I originally had in that method:

```python
        # Load pre-commands available to run
        pre_command_hooks = self.get_hook_results("pre_commands")

        for pre_command in pre_command_hooks:
            if command in pre_command.run_for:
                try:
                    pre_command.action(args)
                except Exception as exc:
                    error, *_ = traceback.format_exception_only(type(exc), exc)
                    error = error.strip()

                    log.error(
                        f'Pre-command action for the plugin "{pre_command.name}" failed with: '
                        f"{error}"
                    )
```

It would basically just catch everything and then log it. Should we consider re-introducing this. @dholth's concern about not being able to uninstall a faulty plugin does concern me 😅.

Another way around this would be to introduce a new setting called called `disabled_plugins`. You would only need to de-activate faulty plugins. We could also perhaps automatically disable a plugin if it were to raise an uncaught exception somewhere. Somewhere in the error message, we could tell the user what happened.

Definitely a can of worms but worth discussing.

Reply:
@jezdez and @dholth,

I was able to update this pull request today with the addition of a `disabled_plugins` setting. Please let me know what you think and whether this could be a solution to the problem of "misbehaving plugins".

Reply:
As I mentioned in person, I don't think we should catch all errors of plugins and ignore them and move on. The functioning of a plugin may be required for the user (imagine a plugin setting auth credentials or something), so it's important to raise any issue with similar to other errors that conda shows to users. Plugins are part of the conda runtime environment after all and need to be treated that way as well.

Let's add a global `--no-plugins` CLI options as the escape hatch Daniel alluded to. And separate that into an own PR related to https://github.com/conda/conda/issues/12149 to not make review harder here.

I'm a little concerned that `disabled_plugins`, is going to be a painful user interface for end users, since there is no presedence so far to listing plugin names (will it be plugin names, or dotted paths to implementation hooks?). Basically, it's too low-level for regular users IMO.

Reply:
I summed up our discussion on https://github.com/conda/conda/issues/12149 . So, let's move that conversation over there for now. I am considering this thread resolved.